### PR TITLE
Some vet tools fixes

### LINKED ISSUE #686: vet tool
Reports from particular vet tools:
### nilness
```
# gopkg.in/src-d/go-mysql-server.v0/sql/plan
sql/plan/limit_test.go:85:19: impossible condition: non-nil == nil
```

### shadow
```
# gopkg.in/src-d/go-mysql-server.v0/mem
mem/table_test.go:197:8: declaration of "err" shadows declaration at line 193
# gopkg.in/src-d/go-mysql-server.v0/sql
sql/catalog.go:167:9: declaration of "err" shadows declaration at line 164
# gopkg.in/src-d/go-mysql-server.v0/auth_test
auth/common_test.go:104:4: declaration of "req" shadows declaration at line 96
auth/common_test.go:106:8: declaration of "err" shadows declaration at line 98
auth/common_test.go:199:4: declaration of "req" shadows declaration at line 191
auth/common_test.go:201:8: declaration of "err" shadows declaration at line 193
# gopkg.in/src-d/go-mysql-server.v0/sql/analyzer
sql/analyzer/assign_indexes.go:268:3: declaration of "result" shadows declaration at line 91
sql/analyzer/prune_columns.go:20:3: declaration of "n" shadows declaration at line 13
sql/analyzer/pushdown.go:62:2: declaration of "filters" shadows declaration at line 10
sql/analyzer/pushdown.go:169:7: declaration of "err" shadows declaration at line 77
sql/analyzer/pushdown.go:171:13: declaration of "err" shadows declaration at line 77
# gopkg.in/src-d/go-mysql-server.v0/sql/expression
sql/expression/comparison.go:74:4: declaration of "left" shadows declaration at line 71
sql/expression/comparison.go:74:10: declaration of "right" shadows declaration at line 71
sql/expression/comparison.go:84:4: declaration of "left" shadows declaration at line 71
sql/expression/comparison.go:84:10: declaration of "right" shadows declaration at line 71
sql/expression/comparison.go:93:3: declaration of "left" shadows declaration at line 71
sql/expression/comparison.go:93:9: declaration of "right" shadows declaration at line 71
sql/expression/like.go:62:6: declaration of "err" shadows declaration at line 46
sql/expression/unresolved.go:133:3: declaration of "c" shadows declaration at line 132
# gopkg.in/src-d/go-mysql-server.v0/sql/expression/function
sql/expression/function/ceil_round_floor.go:192:10: declaration of "err" shadows declaration at line 180
sql/expression/function/ceil_round_floor.go:283:8: declaration of "err" shadows declaration at line 275
sql/expression/function/concat.go:70:3: declaration of "arg" shadows declaration at line 69
sql/expression/function/concat_ws.go:68:3: declaration of "arg" shadows declaration at line 67
sql/expression/function/time_test.go:348:2: declaration of "clock" shadows declaration at line 539
# gopkg.in/src-d/go-mysql-server.v0/sql/parse
sql/parse/lock.go:42:7: declaration of "err" shadows declaration at line 35
sql/parse/parse.go:180:9: declaration of "err" shadows declaration at line 165
sql/parse/parse.go:1046:9: declaration of "err" shadows declaration at line 1035
sql/parse/show_create.go:45:7: declaration of "err" shadows declaration at line 39
sql/parse/show_create.go:74:7: declaration of "err" shadows declaration at line 66
sql/parse/show_create.go:78:7: declaration of "err" shadows declaration at line 66
sql/parse/show_create.go:84:5: declaration of "err" shadows declaration at line 66
# gopkg.in/src-d/go-mysql-server.v0/sql/index/pilosa
sql/index/pilosa/driver.go:185:5: declaration of "err" shadows declaration at line 181
sql/index/pilosa/driver.go:192:8: declaration of "err" shadows declaration at line 181
sql/index/pilosa/driver.go:210:6: declaration of "err" shadows declaration at line 181
sql/index/pilosa/driver.go:256:10: declaration of "err" shadows declaration at line 250
sql/index/pilosa/driver.go:270:5: declaration of "err" shadows declaration at line 250
sql/index/pilosa/driver.go:302:21: declaration of "err" shadows declaration at line 250
sql/index/pilosa/driver.go:312:11: declaration of "err" shadows declaration at line 302
sql/index/pilosa/driver.go:338:6: declaration of "err" shadows declaration at line 250
sql/index/pilosa/driver.go:359:5: declaration of "err" shadows declaration at line 351
sql/index/pilosa/driver.go:404:14: declaration of "err" shadows declaration at line 351
sql/index/pilosa/index.go:306:10: declaration of "err" shadows declaration at line 299
sql/index/pilosa/index.go:352:10: declaration of "err" shadows declaration at line 345
sql/index/pilosa/lookup.go:353:5: declaration of "err" shadows declaration at line 340
sql/index/pilosa/lookup.go:458:8: declaration of "err" shadows declaration at line 446
sql/index/pilosa/mapping.go:175:6: declaration of "err" shadows declaration at line 169
sql/index/pilosa/driver_test.go:195:12: declaration of "err" shadows declaration at line 171
sql/index/pilosa/driver_test.go:202:10: declaration of "err" shadows declaration at line 195
sql/index/pilosa/driver_test.go:872:7: declaration of "err" shadows declaration at line 846
sql/index/pilosa/driver_test.go:885:8: declaration of "err" shadows declaration at line 846
sql/index/pilosa/driver_test.go:939:8: declaration of "err" shadows declaration at line 905
# gopkg.in/src-d/go-mysql-server.v0/sql/plan
sql/plan/create_index.go:333:7: declaration of "i" shadows declaration at line 325
sql/plan/innerjoin.go:273:7: declaration of "err" shadows declaration at line 270
sql/plan/innerjoin.go:290:9: declaration of "err" shadows declaration at line 270
sql/plan/resolved_table.go:76:8: declaration of "err" shadows declaration at line 73
sql/plan/resolved_table.go:98:6: declaration of "err" shadows declaration at line 96
sql/plan/limit_test.go:41:5: declaration of "testingTableSize" shadows declaration at line 16
sql/plan/limit_test.go:48:5: declaration of "testingTableSize" shadows declaration at line 16
sql/plan/limit_test.go:55:5: declaration of "testingTableSize" shadows declaration at line 16
sql/plan/limit_test.go:62:5: declaration of "testingTableSize" shadows declaration at line 16
sql/plan/showvariables_test.go:22:11: declaration of "err" shadows declaration at line 19
sql/plan/showvariables_test.go:53:11: declaration of "err" shadows declaration at line 50
sql/plan/showwarnings_test.go:26:11: declaration of "err" shadows declaration at line 22
```

### PR DESCRIPTION
**Pull request by juanjux**
Fixes #686 except for a false positive (bug?) in the nilness tool.

### CODE REVIEW

File: auth/common_test.go [hunk 1]
```diff
@@ -196,20 +196,20 @@ func testAudit(
 
 	for _, c := range tests {
 		t.Run(c.user, func(t *testing.T) {
-			req := require.New(t)
+			r := require.New(t)
 
-			db, err := dsql.Open("mysql", connString(c.user, ""))
-			req.NoError(err)
-			_, err = db.Query(c.query)
+			db, e := dsql.Open("mysql", connString(c.user, ""))
```
Comment (mcuadros) [inline auth/common_test.go]: why `e`?

Comment (juanjux) [reply auth/common_test.go]: To shut up `go vet` about reusing `err` in the enclosing context.

Comment (juanjux) [reply auth/common_test.go]: `var db *dsql.DB` actually. It's one line more, but ok, `err` is certainly more easy to human-parse than just `e`.

File: sql/catalog.go [hunk 2]
```diff
@@ -178,8 +178,8 @@ func (c *Catalog) UnlockTables(ctx *Context, id uint32) error {
 			table, err := c.dbs.Table(db, t)
 			if err == nil {
 				if lockable, ok := table.(Lockable); ok {
-					if err := lockable.Unlock(ctx, id); err != nil {
-						errors = append(errors, err.Error())
+					if e := lockable.Unlock(ctx, id); e != nil {
```
Comment (ajnavarro) [inline sql/catalog.go]: err

**Comment by juanjux** [comment]
@ajnavarro check the last commit (I'll rebase it later), I've removed most of the `e`s but for a couple cases in `catalog.go` and `resolved_table.go` where the enclosing error variable is being used later with a value updated in the current context and thus we either keep the `e`  or  the warning (and the danger of somebody not typing `:=` and overwriting the enclosing one on a refactor).

**Pull request merged**